optimum/habana/transformers/models/opt/modeling_opt.py

```diff
@@ -0,0 +1,516 @@
+import random
+from typing import List, Optional, Tuple, Union
+
+import torch
+from torch.nn import CrossEntropyLoss
+from transformers.modeling_outputs import BaseModelOutputWithPast, CausalLMOutputWithPast
+from transformers.models.opt.modeling_opt import OPTForCausalLM, logger
+
+
+class GaudiOPTLearnedPositionalEmbedding(torch.nn.Embedding):
+    def __init__(self, num_embeddings: int, embedding_dim: int):
+        self.offset = 2
+        super().__init__(num_embeddings + self.offset, embedding_dim)
+
+    def forward(
```

Comment:
this could be the best approach, I will do it

Reply:
@regisss in the end of `OPTLearnedPositionalEmbedding.forward` there is a `super().forward` that requires `__class__ cell` but not found
![image](https://github.com/huggingface/optimum-habana/assets/80079571/9be20aba-a817-4699-a673-388e77aa56ae)
any ideas?

Reply:
one things is that `GaudiOPTLearnedPositionalEmbedding` is inherit from `torch.nn.Embedding` not `OPTLearnedPositionalEmbedding`, so the `__init__` cannot be removed and because there is a `super()`, I think we cannot make `GaudiOPTLearnedPositionalEmbedding` inherit from `OPTLearnedPositionalEmbedding`

Reply:
My bad I don't know why I thought it was inheriting from `OPTLearnedPositionalEmbedding`, sorry ^^
Let's keep it that way then!

Reply:
found a solution [here](https://stackoverflow.com/questions/18117974/calling-a-parents-parents-method-which-has-been-overridden-by-the-parent), it works, we could remote `__init__`